chain/chain/src/block_processing_utils.rs

```diff
@@ -29,8 +29,6 @@ pub(crate) struct BlockPreprocessInfo {
     pub(crate) apply_chunks_done: Arc<OnceCell<()>>,
     /// This is used to calculate block processing time metric
     pub(crate) block_start_processing_time: Instant,
-    /// Whether needs to create a state snapshot after processing this block.
-    pub(crate) need_state_snapshot: bool,
```

Comment:
We don't need snapshot details in BlockPreprocessInfo as it's only used in the preprocess function later (and not after post processing)